### PR TITLE
Catch template rendering error early to avoid app crashing

### PR DESCRIPTION
**Pull request by CFarcy**

### CODE REVIEW

**Review by jrmi — APPROVED** [review]

**Review by jrmi — DISMISSED** [review]

**Review by jrmi — APPROVED** [review]

File: src/components/ErrorBoundary.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+export default class ErrorBoundary extends React.Component {
+  constructor (props) {
+    super(props);
+    this.state = { hasError: false };
+  }
```
Comment (dtrucs) [inline src/components/ErrorBoundary.js:7]: Why not simply?
```suggestion
  state = { hasError: false };
```

File: src/modules/Template/Template/Template.js [hunk 2]
```diff
@@ -23,36 +23,47 @@ export const Template = ({
     ? env.renderString(template, props)
     : content);
 
-  const processNodeDefinitions = new ProcessNodeDefinitions(React);
+  const processingInstructions = useMemo(() => {
+    const processNodeDefinitions = new ProcessNodeDefinitions(React);
+    const customInstructions = customComponents.map(spec => {
+      if (spec.processNode) {
+        return spec;
+      }
```
Comment (dtrucs) [inline src/modules/Template/Template/Template.js:31]: Do you really want a `customInstructions` array with potential holes? Maybe a `filter` method should be more appropriate?

File: src/components/ErrorBoundary.js (hunk 1 above)
Comment (CFarcy) [reply src/components/ErrorBoundary.js:7]: I just copy/past the doc https://reactjs.org/docs/error-boundaries.html, but I can change it if not needed

File: src/modules/Template/Template/Template.js (hunk 2 above)
Comment (CFarcy) [reply src/modules/Template/Template/Template.js:31]: It either return the `spec` or an object construct around `spec`

Comment (jrmi) [reply src/modules/Template/Template/Template.js:31]: I don't see where is the hole neither?

Comment (dtrucs) [reply src/modules/Template/Template/Template.js:31]: wow that's my bad, at first glance I thought the `map` ended after that `if`.

**Review by dtrucs — APPROVED** [review]

**Pull request merged**